### PR TITLE
Support USDC deposit to savings through PSMActions

### PR DESCRIPTION
**Pull request by yivlad**
Closes FRO-739

### CODE REVIEW

**Comment by krzkaczor** [comment]
> Powered by Maker. No slippage & fees for DAI.

The badge should mention USDC not DAI 😄

File: packages/app/src/features/actions/logic/useCreateActions.ts [hunk 1]
```diff
@@ -167,10 +168,13 @@ export function useCreateActions(objectives: Objective[]): Action[] {
       }
 
       case 'nativeSDaiDeposit': {
+        const spender =
+          objective.token.symbol === 'USDC' ? psmActionsConfig.address[mainnet.id] : objective.sDai.address
```
Comment (oskarvu) [inline packages/app/src/features/actions/logic/useCreateActions.ts:172]: Same as earlier, compare with `TokenSymbol('USDC')`

File: packages/app/src/features/dialogs/savings/common/components/maker-transaction-overview/components/RouteItem.tsx [hunk 2]
```diff
@@ -8,21 +8,33 @@ export interface RouteItemProps {
   item: RouteItemType
   index: number
   isLast: boolean
+  verticalRouteDisplay: boolean
```
Comment (oskarvu) [inline packages/app/src/features/dialogs/savings/common/components/maker-transaction-overview/components/RouteItem.tsx:11]: nit: `displayVertically`

File: packages/app/src/features/savings/components/savings-dai/SavingsDAI.tsx [hunk 3]
```diff
@@ -51,7 +51,7 @@ export function SavingsDAI({
       <div className="flex flex-col items-center gap-1 sm:gap-2">
         <div
           className="flex flex-row items-end justify-center slashed-zero tabular-nums"
-          data-testId={testIds.savings.sDaiBalanceInDai}
+          data-testid={testIds.savings.sDaiBalanceInDai}
```
Comment (oskarvu) [inline packages/app/src/features/savings/components/savings-dai/SavingsDAI.tsx:54]: 👍

**Review by oskarvu — APPROVED** [review]

File: packages/app/src/features/actions/logic/useCreateActions.ts [hunk 4]
```diff
@@ -1,11 +1,12 @@
 import { maxUint256 } from 'viem'
 
 import { getNativeAssetInfo } from '@/config/chain/utils/getNativeAssetInfo'
-import { wethGatewayAddress } from '@/config/contracts-generated'
+import { psmActionsConfig, wethGatewayAddress } from '@/config/contracts-generated'
 import { useContractAddress } from '@/domain/hooks/useContractAddress'
 import { useOriginChainId } from '@/domain/hooks/useOriginChainId'
 import { BaseUnitNumber, NormalizedUnitNumber } from '@/domain/types/NumericValues'
 
+import { mainnet } from 'viem/chains'
```
[the remark this answers is not in the document]
Comment (krzkaczor) [reply packages/app/src/features/actions/logic/useCreateActions.ts:9]: maybe browse biome issues, we can't be the only ones having issues with this

**Review by krzkaczor — APPROVED** [review]

File: packages/app/src/features/actions/logic/useCreateActions.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (oskarvu) [reply packages/app/src/features/actions/logic/useCreateActions.ts:9]: https://github.com/biomejs/biome/issues/3177 Looks like improvements are planned.

**Pull request merged**